tests/ui/diverging-fallback-option.rs

```diff
@@ -1,14 +0,0 @@
-//@ run-pass
-
-#![allow(warnings)]
-
-// Here the type of `c` is `Option<?T>`, where `?T` is unconstrained.
-// Because there is data-flow from the `{ return; }` block, which
-// diverges and hence has type `!`, into `c`, we will default `?T` to
-// `!`, and hence this code compiles rather than failing and requiring
-// a type annotation.
```

Comment:
Suggestion: can you actually retain this comment over

```
//! Checks that when creating `Option` from a diverging expression,
//! the type parameter defaults to `!` without requiring explicit annotations.
```

The inference vars here I think is important context.